apps/mobile/src/components/HexDataDisplay/HexDataDisplay.tsx

```diff
@@ -49,7 +49,7 @@ export const HexDataDisplay = ({
   return (
     <InfoSheet title={title} info={data}>
       <View flexDirection="row" alignItems="center" gap={gap}>
-        <Text>{shortenText(data, characterLimit)}</Text>
+        <Text>{data.length > characterLimit ? shortenText(data, characterLimit) : data}</Text>
```

Comment:
shouldn't it be done inside the `shortenText` function? Otherwise we will need to do it every place we call the shortenText function